packages/js/src/ui/helpers/use-style.tsx

```diff
@@ -0,0 +1,32 @@
+import { createMemo, createSignal, onMount } from 'solid-js';
+import { Elements, useAppearance } from '../context';
+import { cn } from './utils';
+
+export const useStyle = () => {
+  const appearance = useAppearance();
+  const [isServer, setIsServer] = createSignal(true);
+
+  onMount(() => {
+    setIsServer(false);
+  });
+
+  const styleFuncMemo = createMemo(() => (className: string, descriptor?: keyof Elements | keyof Elements[]) => {
+    const appearanceClassname =
+      descriptor && typeof appearance.elements?.[descriptor] === 'string'
+        ? (appearance.elements?.[descriptor] as string) || ''
+        : '';
+
+    const descriptors = (Array.isArray(descriptor) ? descriptor : [descriptor]).map((des) => `nv-${des}`);
+    const cssInJsClasses =
+      !!descriptors.length && !isServer() ? descriptors.map((des) => appearance.descriptorToCssInJsClass[des]) : [];
+
+    return cn(
+      ...descriptors,
+      className, // default styles
```

Comment:
This is just the order that the classes will be shown on the element. We want the descriptors first so that it's the first thing the dev sees. Order doesn't really matter.